addons/web/static/src/views/fields/translation_dialog.js

```diff
@@ -74,17 +75,27 @@ export class TranslationDialog extends Component {
     async onSave() {
         const translations = {};
 
-        this.terms.map((term) => {
-            const updatedTermValue = this.updatedTerms[term.id];
-            if (term.id in this.updatedTerms && term.value !== updatedTermValue) {
-                if (this.props.showSource) {
+        const resetLangs = [];
+        this.props.languages.forEach(([language, languageName]) => {
+            if (
+                !this.state.terms.some(
+                    (t) => t.lang === language && (t.isTranslated || !t.isModified)
+                )
+            ) {
+                resetLangs.push(language);
+            }
+        });
```

Comment:
Not to impose some rule but I believe this is much readable if it's a for-loop. Something like:
```js
for (const [language] of this.props.languages) {
  // ... mutate resetLangs here.
}
```

Also the use of `map` below (`this.state.terms.map(...`), convert it to for-loop.